src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -202,16 +202,16 @@ public void onStarted(Run run, TaskListener listener) {
             }
 
             Long waitingMs;
-            try {
-                Queue queue = getQueue();
-                Queue.Item item = queue.getItem(run.getQueueId());
+            Queue queue = getQueue();
+            Queue.Item item = queue.getItem(run.getQueueId());
+            if (item != null){
                 waitingMs = (DatadogUtilities.currentTimeMillis() - item.getInQueueSince());
                 PipelineQueueInfoAction queueInfoAction = run.getAction(PipelineQueueInfoAction.class);
                 if (queueInfoAction != null) {
                     // this needs to be set before BuildData is created, as BuildData will use this value
                     queueInfoAction.setQueueTimeMillis(waitingMs);
                 }
-            } catch (NullPointerException e) {
+            } else {
                 // item.getInQueueSince() may raise a NPE if a worker node is spinning up to run the job.
```

Comment:
Now that the catch is replaced with a null-check, the comment should be updated
```suggestion
                // item may be null if a worker node is spinning up to run the job.
```